extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/controlBar.js

```diff
@@ -0,0 +1,506 @@
+    var playIcon = "";
+    if (this.props.playerState == CONSTANTS.STATE.PLAYING) {
+      playIcon = "pause";
+    } else if (this.props.playerState == CONSTANTS.STATE.END) {
+      playIcon = "replay";
+    } else {
+      playIcon = "play";
+    }
+
+    var volumeIcon = (this.props.controller.state.volumeState.muted ? "volumeOff" : "volume");
+
+    var fullscreenIcon = "";
+    if (this.props.controller.state.fullscreen) {
+      fullscreenIcon = "compress"
+    }
+    else {
+      fullscreenIcon = "expand";
+    }
+
+    var totalTime = 0;
+    if (this.props.duration == null || typeof this.props.duration == 'undefined' || this.props.duration == ""){
+      totalTime = Utils.formatSeconds(0);
+    }
+    else {
+      totalTime = Utils.formatSeconds(this.props.duration);
+    }
+
+    var volumeBars = [];
+    for (var i=0; i<10; i++) {
+      //create each volume tick separately
+      var turnedOn = this.props.controller.state.volumeState.volume >= (i+1) / 10;
+      var volumeClass = ClassNames({
+        "oo-volume-bar": true,
+        "oo-on": turnedOn
+      });
+      var barStyle = turnedOn ? {backgroundColor: this.props.skinConfig.controlBar.volumeControl.color ? this.props.skinConfig.controlBar.volumeControl.color : this.props.skinConfig.general.accentColor} : null;
+
+      volumeBars.push(<a data-volume={(i+1)/10} className={volumeClass} key={i}
+        style={barStyle}
+        onClick={this.handleVolumeClick}></a>);
+    }
+
+    var volumeSlider = <div className="oo-volume-slider"><Slider value={parseFloat(this.props.controller.state.volumeState.volume)}
+                        onChange={this.changeVolumeSlider}
+                        className={"oo-slider oo-slider-volume"}
+                        itemRef={"volumeSlider"}
+                        minValue={"0"}
+                        maxValue={"1"}
+                        step={"0.1"}/></div>;
+
+    var volumeControls;
+    if (!this.isMobile){
+      volumeControls = volumeBars;
+    }
+    else {
+      volumeControls = this.props.controller.state.volumeState.volumeSliderVisible ? volumeSlider : null;
+    }
+
+    var playheadTime = isFinite(parseInt(this.props.currentPlayhead)) ? Utils.formatSeconds(parseInt(this.props.currentPlayhead)) : null;
+    var isLiveStream = this.props.isLiveStream;
+    var durationSetting = {color: this.props.skinConfig.controlBar.iconStyle.inactive.color};
+    var timeShift = this.props.currentPlayhead - this.props.duration;
+    // checking timeShift < 1 seconds (not == 0) as processing of the click after we rewinded and then went live may take some time
+    var isLiveNow = Math.abs(timeShift) < 1;
+    var liveClick = isLiveNow ? null : this.handleLiveClick;
+    var playheadTimeContent = isLiveStream ? (isLiveNow ? null : Utils.formatSeconds(timeShift)) : playheadTime;
+    var totalTimeContent = isLiveStream ? null : <span className="oo-total-time">{totalTime}</span>;
+
+    // TODO: Update when implementing localization
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)